packages/vscode-extension/src/controls/sampleGallery/sampleFilter.tsx

```diff
@@ -16,49 +16,101 @@ import {
 } from "../../telemetry/extTelemetryEvents";
 import { Commands } from "../Commands";
 import { Grid } from "../resources";
-import { SampleFilterProps, SampleFilterState, SampleInfo } from "./ISamples";
+import { SampleFilterProps } from "./ISamples";
+
+export default class SampleFilter extends React.Component<SampleFilterProps, unknown> {
+  private sampleTypes = ["Tab", "Bot", "Message extension"];
+  private sampleLanguages = ["TS", "JS"];
+  private sampleTechniques = ["Azure", "Adaptive Cards", "SSO", "SPFx", "Outlook", "Graph"];
```

Comment:
Will we need to extend 'sampleTechniques' field when more samples are onboarded?

Reply:
Yes, these are the options that I chose firstly, I think this could be updated later according to user behavior. We can keep the most representative ones in the list. Also @summzhan could help to review this list.

Reply:
Yes, I think we'd better make it general and complete, or there'll be dependency between samples and TTK. cc @summzhan

Reply:
Do the values of all these tags, not only techniques, but also type and languages, could be stored in some config file but not hard code? We should as flexible as possible to add new values. It's quite hard to predict new techniques/types appear in future...., at least the type tag is about to add a new value which is copilot plugin in near future right?

Reply:
@summzhan Do you mean to store the option values in an online config file like `sample-config.json`?

Reply:
Updated code to add filter options in sample config file after offline discussion. cc @HuihuiWu-Microsoft @summzhan